### PR TITLE
ddl: check table charset option

### LINKED ISSUE #5743: Table charset option need a valid check
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
```
CREATE TABLE test4 (
id bigint(20) NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=abcded;
show create table test4;
```

2. What did you expect to see?

Raise an error, because no charset called "abcded".

3. What did you see instead?

No error occur, and it shows the following
```
+-------+-----------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                              |
+-------+-----------------------------------------------------------------------------------------------------------+
| test4 | CREATE TABLE `test4` (
  `id` bigint(20) NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=abcded COLLATE=utf8_bin |
+-------+-----------------------------------------------------------------------------------------------------------+
```
4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
```
Release Version: v1.1.0-alpha.1-26-g04960d4f
Git Commit Hash: 04960d4fe999ab6b1c222c5528ffdfcf59b6cd87
Git Commit Branch: master
UTC Build Time:  2018-01-29 07:49:41
GoVersion:  go version go1.9.2 darwin/amd64
```

### PR DESCRIPTION
**Pull request by yangwenmai**
I fixed #5743

### CODE REVIEW

**Comment by coocood** [comment]
@yangwenmai 
The check should be done before the DDL Job is created, please take a look at the `CreateTable` method in `ddl_api.go`

**Comment by coocood** [comment]
And this fix makes an old test case failed, we should update the test case.

File: executor/show_test.go [hunk 1]
```diff
@@ -310,12 +310,9 @@ func (s *testSuite) TestShow(c *C) {
...
 
 	tk.MustExec(`drop table if exists t`)
-	tk.MustExec(`create table t (a int) default charset=abcdefg`)
-	tk.MustQuery(`show create table t`).Check(testutil.RowsWithSep("|",
-		"t CREATE TABLE `t` (\n"+
-			"  `a` int(11) DEFAULT NULL\n"+
-			") ENGINE=InnoDB DEFAULT CHARSET=abcdefg",
-	))
+
+	_, err := tk.Exec("create table t (a int) default charset=abcdefg")
```
Comment (jackysp) [inline executor/show_test.go]: I think this test case should move away from show_test.go . Maybe ddl_db_test.go is a good place for it.

File: executor/show_test.go [hunk 2]
```diff
@@ -310,12 +310,9 @@ func (s *testSuite) TestShow(c *C) {
 	))
 
 	tk.MustExec(`drop table if exists t`)
-	tk.MustExec(`create table t (a int) default charset=abcdefg`)
```
Comment (coocood) [inline executor/show_test.go]: We don't need to remove the test case, we can change the charset to a valid one like `utf8mb4`.

Comment (yangwenmai) [reply executor/show_test.go]: Already included the correct test cases in https://github.com/pingcap/tidb/blob/master/executor/show_test.go#L305

File: executor/show_test.go (hunk 1 above)
Comment (yangwenmai) [reply executor/show_test.go]: Yes, I think so.

**Comment by jackysp** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

@yangwenmai 
Thank you for your contribution!

**Pull request merged**